buckets/local/pull.go

```diff
@@ -138,9 +141,12 @@ looop:
 			}
 		}
 		eg, gctx := errgroup.WithContext(context.Background())
+		lim := make(chan struct{}, MaxPullConcurrency)
```

Comment:
When playing around with the docs bucket, I ran into the `too many files open` error since pull was trying to pull a huge amount of file. This limits how many are pulled concurrently at a given time.